docs/template_plugin/tests/functional/op_reference/scatter_elements_update.cpp

```diff
@@ -0,0 +1,177 @@
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <tuple>
+
+#include "base_reference_test.hpp"
+
+using namespace reference_tests;
+using namespace ngraph;
+using namespace InferenceEngine;
+
+namespace {
+struct ScatterElementsUpdateParams {
+    ScatterElementsUpdateParams(const Tensor& paramData,
+                                const Tensor& paramIndices,
+                                const Tensor& paramUpdates,
+                                const Tensor& paramAxis,
+                                const Tensor& paramExpected)
+        : input(paramData),
+          indices(paramIndices),
+          updates(paramUpdates),
+          axis(paramAxis),
+          expected(paramExpected) {}
+
+    Tensor input;
+    Tensor indices;
+    Tensor updates;
+    Tensor axis;
+    Tensor expected;
+};
+
+class ReferenceScatterElementsUpdateLayerTest : public testing::TestWithParam<ScatterElementsUpdateParams>,
+                                       public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.input.data,
+                     params.indices.data,
+                     params.updates.data,
+                     params.axis.data};
+        refOutData = {params.expected.data};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<ScatterElementsUpdateParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "data_sh=" << param.input.shape;
+        result << "_data_pr=" << param.input.type;
+        result << "_indx_sh=" << param.indices.shape;
+        result << "_indx_pr=" << param.indices.type;
+        result << "_updt_sh=" << param.updates.shape;
+        result << "_updt_pr=" << param.updates.type;
+        result << "_axis_sh=" << param.axis.shape;
+        result << "_axis_pr=" << param.axis.type;
+        return result.str();
+    }
+private:
+    static std::shared_ptr<Function> CreateFunction(const ScatterElementsUpdateParams& params) {
+        const auto A = std::make_shared<op::Parameter>(params.input.type, params.input.shape);
+        const auto B = std::make_shared<op::Parameter>(params.indices.type, params.indices.shape);
+        const auto C = std::make_shared<op::Parameter>(params.updates.type, params.updates.shape);
+        const auto D = std::make_shared<op::Parameter>(params.axis.type, params.axis.shape);
+        auto scatterElts = std::make_shared<op::ScatterElementsUpdate>(A, B, C, D);
+        return std::make_shared<Function>(NodeVector{scatterElts}, ParameterVector{A, B, C, D});
+    }
+};
+
+TEST_P(ReferenceScatterElementsUpdateLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+template <element::Type_t ET, element::Type_t ET_IND>
+std::vector<ScatterElementsUpdateParams> generateScatterParams(const element::Type& type,
```

Comment:
You can remove input parameters `type` and `type_int` from `generateScatterParams` function.
When creating tensors use instead `element::Type(T)` and `element::Type(T_INT)` respectively